File: java/leetcode-explore/highFrequency/hash/TwoSum.java
package highFrequency.hash;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> memo = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(memo.containsKey(target - nums[i])){
                return new int[]{i, memo.get(target - nums[i])};
            }
            memo.put(nums[i], i);
        }
        return null;
    }
}

File: go/tencent/array_and_str/two_sum.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{3, 2, 4}, 6))
}

func twoSum(nums []int, target int) []int {
	memo := make(map[int]int)

	for idx, val := range nums {
		if _, ok := memo[target-val]; ok {
			return []int{memo[target-val], idx}
		}
		memo[val] = idx
	}
	return []int{-1, -1}
}
